thorclient/api_test.go

```diff
@@ -0,0 +1,381 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package thorclient
+
+import (
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/require"
+	"github.com/vechain/thor/v2/api/accounts"
+	"github.com/vechain/thor/v2/api/blocks"
+	"github.com/vechain/thor/v2/api/debug"
+	"github.com/vechain/thor/v2/api/events"
+	"github.com/vechain/thor/v2/api/node"
+	"github.com/vechain/thor/v2/comm"
+	"github.com/vechain/thor/v2/genesis"
+	"github.com/vechain/thor/v2/logdb"
+	"github.com/vechain/thor/v2/test/datagen"
+	"github.com/vechain/thor/v2/test/testchain"
+	"github.com/vechain/thor/v2/thor"
+	"github.com/vechain/thor/v2/tx"
+	"github.com/vechain/thor/v2/txpool"
+
+	// Force-load the tracer native engines to trigger registration
+	_ "github.com/vechain/thor/v2/tracers/js"
+	_ "github.com/vechain/thor/v2/tracers/logger"
+)
+
+const (
+	gasLimit   = 30_000_000
+	logDBLimit = 1_000
+)
+
+func initAPIServer(t *testing.T) (*testchain.Chain, *httptest.Server) {
```

Comment:
If I am understanding correctly, here is to build a full equivalent of `thor/cmd/node(solo)` but the main package called `testchain` with some component not introduced and some component was built in place(eg. logsdb), what's your thoughts?

Reply:
Yes originally that was the idea, however after investigating I gave up on that idea because the thor-solo does a lot more than responding to API requests.
This test API server is mainly an extension of the pattern used for testing the `api` package.

Reply:
Following your thoughts, I might consider `logsdb` should be equivalent to `bft`.

Reply:
Yep, I get what you mean. I made the changes to put the logdb in the `testchain`
wdyt ?